### DISCUSSION TITLE
tcpdump hex: UTF8 in ASCII Ã¶

### DISCUSSION THREAD

**Discussion by mkwba** [OpenIDC/mod_auth_openidc#1017] [Q&A]
Hello,
we are using a sonarqube behind a apache_proxy with this mod.
If somebody login with an "ö or ä,ü" the sonarqube shows "Ã¶" instead of "ö".

With strace on the apache pid the "ö" come from the oidc-provider with "\303\266" into the apache.
A tcpdump -X between sonarqube and the apache shows **c3b6** where the "ö" is.

If i throw this into an Hex to ASCII-Converter i can see "Ã¶" instead of "ö".
(Example)
656f c3b6 6c72
results in: eoÃ¶lr

If i convert Hexadezimal to UTF8 now and set in the hex-value a whitespace like "c3 b6" i can see the ö.
Is this a bug in the mod or a missconfiguration on our side?
I see other posts here in discussion but have not the knowledge to identify our problem in that.
thanks

**Comment by zandbelt** [comment]
Assuming sonarcube relies (or can be made to rely) on the standardized-by-RFC header format  ISO-8859-1, you could try with the most recent version of the module and set:
```
OIDCPassClaimsAs headers latin1
```
see: https://github.com/zmartzone/mod_auth_openidc/blob/v2.4.12.3/auth_openidc.conf#L810-L818
and: https://github.com/zmartzone/mod_auth_openidc/discussions/957

**Comment by mkwba** [comment]
Thank you for your reply!
(I've updated the mod to the latest version now)

After adding the line above, the ASCII-Format for 
`OIDC_CLAIM_name`
 now is correct:
`f6`

But the headers:
`X-Forwarded-User-Fullname` (Groups, User-Email) are all
`(null)` now.

`X-Forwarded-Groups:.(null)`
`X-Forwarded-User-Email:.(null)`
`X-Forwarded-User-Fullname:.(null)`

**Reply by zandbelt** [reply]
I'm not sure who's supposed to set the `X-Forwarded-*` headers, but it is not mod_auth_openidc; perhaps you're using mod_headers or similar?

**Reply by mkwba** [reply]
Sonarqube requires the headers.
Yes we are using the apache mod_headers.

In my site-conf is a translation: 
    `RequestHeader set "X-Forwarded-Groups" %{OIDC_CLAIM_groups}e`
    `RequestHeader set "X-Forwarded-User-Email" %{OIDC_CLAIM_email}e`
    `RequestHeader set "X-Forwarded-User-Fullname" %{OIDC_CLAIM_name}e`

I have also tried changing "AddDefaultCharset utf-8" to latin1 but no effect.

**Reply by zandbelt** [reply]
so use:
```
OIDCPassClaimsAs both latin1
```

**Reply by mkwba** [reply]
This is the solution! Thank you very much!
